### ISSUE TITLE
RTX 3090 training custom data stuck on first epoch

### ISSUE THREAD

**Issue by er778899789** [tensorflow/tensorflow#49826]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v1.12.1-54779-g1236f723c6d 2.6.0-dev20210413
- Python version: 3.8.8
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: CUDA 11.1/CUDNN 8.1.0
- GPU model and memory: Nvidia RTX 3090 64GB

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
I used to run the code on Colab and it run successfully without any errors. I want to copy this code on my new computer.
But when I run the code, it will stuck on the first epoch and never continue.
I want to know what cause this problem, I can't see any errors on the log.

**Describe the expected behavior**
Run the training code successfully with GPU RTX 3090 and tensorflow.

**Standalone code to reproduce the issue**
I just put some important code on here.
```python=
img_size = 256
num_class = len(training_set_new['Label'].unique())
print(num_class)

from tensorflow.keras.applications.efficientnet import EfficientNetB7, preprocess_input

pre_model = EfficientNetB7(weights='imagenet', include_top=False, classes=6,
                        input_shape=(img_size, img_size, 3), drop_connect_rate=0.6)

x = GlobalAveragePooling2D()(pre_model.output)
x = Dense(512, activation='relu')(x)
x = Dropout(0.3)(x)
outputs = Dense(num_class, activation='softmax')(x)


model = Model(inputs=pre_model.inputs, outputs=outputs)
#model.load_weights('./model-logs/DenseNet/basic_model-best-model.h5')

#learning_rate = 0.0001
learning_rate = 0.0001
optimizer = keras.optimizers.Adam(lr=learning_rate)

model.compile(loss='sparse_categorical_crossentropy',
              optimizer=optimizer,
              metrics=['accuracy'])

batch_size = 16
num_steps = len(training_set_new) // batch_size + 1
num_epochs = 1000
```

**Other info / logs** Include any logs or source code that would be helpful to

> 2021-05-27 14:02:41.136307: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library cudart64_110.dll
2021-05-27 14:02:42.704771: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2021-05-27 14:02:42.727386: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library nvcuda.dll
2021-05-27 14:02:42.903739: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1734] Found device 0 with properties:
pciBusID: 0000:2d:00.0 name: GeForce RTX 3090 computeCapability: 8.6
coreClock: 1.695GHz coreCount: 82 deviceMemorySize: 24.00GiB deviceMemoryBandwidth: 871.81GiB/s
2021-05-27 14:02:42.903829: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library cudart64_110.dll
2021-05-27 14:02:42.916577: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library cublas64_11.dll
2021-05-27 14:02:42.916662: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library cublasLt64_11.dll
2021-05-27 14:02:42.923490: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library cufft64_10.dll
2021-05-27 14:02:42.988833: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library curand64_10.dll
2021-05-27 14:02:43.004777: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library cusolver64_11.dll
2021-05-27 14:02:43.012225: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library cusparse64_11.dll
2021-05-27 14:02:43.015654: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library cudnn64_8.dll
2021-05-27 14:02:43.015822: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1872] Adding visible gpu devices: 0
2021-05-27 14:02:43.500449: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1258] Device interconnect StreamExecutor with strength 1 edge matrix:
2021-05-27 14:02:43.500552: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1264]      0
2021-05-27 14:02:43.501406: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1277] 0:   N
2021-05-27 14:02:43.501885: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1418] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 21670 MB memory) -> physical GPU (device: 0, name: GeForce RTX 3090, pci bus id: 0000:2d:00.0, compute capability: 8.6)
1.0    934
3.0    718
0.0    674
5.0    644
4.0    456
2.0    100
Name: Label, dtype: int64
0.0    674
5.0    644
2.0    100
1.0     50
3.0     38
4.0     24
Name: Label, dtype: int64
6
2021-05-27 14:02:46.007355: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1734] Found device 0 with properties:
pciBusID: 0000:2d:00.0 name: GeForce RTX 3090 computeCapability: 8.6
coreClock: 1.695GHz coreCount: 82 deviceMemorySize: 24.00GiB deviceMemoryBandwidth: 871.81GiB/s
2021-05-27 14:02:46.007501: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1872] Adding visible gpu devices: 0
2021-05-27 14:02:46.008117: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1734] Found device 0 with properties:
pciBusID: 0000:2d:00.0 name: GeForce RTX 3090 computeCapability: 8.6
coreClock: 1.695GHz coreCount: 82 deviceMemorySize: 24.00GiB deviceMemoryBandwidth: 871.81GiB/s
2021-05-27 14:02:46.008194: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1872] Adding visible gpu devices: 0
2021-05-27 14:02:46.008529: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1258] Device interconnect StreamExecutor with strength 1 edge matrix:
2021-05-27 14:02:46.008773: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1264]      0
2021-05-27 14:02:46.009054: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1277] 0:   N
2021-05-27 14:02:46.009373: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1418] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 21670 MB memory) -> physical GPU (device: 0, name: GeForce RTX 3090, pci bus id: 0000:2d:00.0, compute capability: 8.6)
C:\Users\Q2785\.conda\envs\Aidea_AOI\lib\site-packages\tensorflow\python\keras\optimizer_v2\optimizer_v2.py:374: UserWarning: The `lr` argument is deprecated, use `learning_rate` instead.
  warnings.warn(
Found 3526 non-validated image filenames.
Found 1530 non-validated image filenames.
2021-05-27 14:02:50.284279: I tensorflow/core/profiler/lib/profiler_session.cc:126] Profiler session initializing.
2021-05-27 14:02:50.284341: I tensorflow/core/profiler/lib/profiler_session.cc:141] Profiler session started.
2021-05-27 14:02:50.284791: I tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1611] Profiler found 1 GPUs
2021-05-27 14:02:50.288669: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library cupti64_112.dll
2021-05-27 14:02:50.360538: I tensorflow/core/profiler/lib/profiler_session.cc:158] Profiler session tear down.
2021-05-27 14:02:50.360749: I tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1743] CUPTI activity buffer flushed
C:\Users\Q2785\.conda\envs\Aidea_AOI\lib\site-packages\tensorflow\python\keras\engine\training.py:1945: UserWarning: `Model.fit_generator` is deprecated and will be removed in a future version. Please use `Model.fit`, which supports generators.
  warnings.warn('`Model.fit_generator` is deprecated and '
C:\Users\Q2785\.conda\envs\Aidea_AOI\lib\site-packages\tensorflow\python\keras\utils\generic_utils.py:494: CustomMaskWarning: Custom mask layers require a config and must override get_config. When loading, the custom mask layer must be passed to the custom_objects argument.
  warnings.warn('Custom mask layers require a config and must override '
2021-05-27 14:02:50.789915: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:180] None of the MLIR Optimization Passes are enabled (registered 2)
Epoch 1/1000
2021-05-27 14:03:15.440492: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library cudnn64_8.dll
2021-05-27 14:03:16.021892: I tensorflow/stream_executor/cuda/cuda_dnn.cc:367] Loaded cuDNN version 8100
2021-05-27 14:03:18.019932: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library cublas64_11.dll
2021-05-27 14:03:18.969248: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library cublasLt64_11.dll
2021-05-27 14:03:20.253134: I tensorflow/stream_executor/cuda/cuda_blas.cc:1838] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.
  1/221 [..............................] - ETA: 1:53:26 - loss: 1.8405 - accuracy: 0.12502021-05-27 14:03:27.764972: I tensorflow/core/profiler/lib/profiler_session.cc:126] Profiler session initializing.
2021-05-27 14:03:27.765052: I tensorflow/core/profiler/lib/profiler_session.cc:141] Profiler session started.

**Comment by er778899789** [comment]
After I close the stuck code, I run the code again.
And I get the following log. Above this part is the same. Only this part different with issue.

>   1/221 [..............................] - ETA: 1:47:42 - loss: 1.7554 - accuracy: 0.31252021-05-27 15:13:22.496873: I tensorflow/core/profiler/lib/profiler_session.cc:126] Profiler session initializing.
2021-05-27 15:13:22.496957: I tensorflow/core/profiler/lib/profiler_session.cc:141] Profiler session started.
  2/221 [..............................] - ETA: 24:31 - loss: 1.7655 - accuracy: 0.2812  2021-05-27 15:13:23.337160: I tensorflow/core/profiler/lib/profiler_session.cc:66] Profiler session collecting data.
2021-05-27 15:13:23.338859: I tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1743] CUPTI activity buffer flushed

**Comment by er778899789** [comment]
I solve the problem. THX

**Comment by Minxiangliu** [comment]
@er778899789 How did you solve it please?

**Comment by yangzhenze1998** [comment]
> After I close the stuck code, I run the code again. And I get the following log. Above this part is the same. Only this part different with issue. After this "CUPTI activity buffer flushed", the code will interrupt.
> 
> > 1/221 [..............................] - ETA: 1:47:42 - loss: 1.7554 - accuracy: 0.31252021-05-27 15:13:22.496873: I tensorflow/core/profiler/lib/profiler_session.cc:126] Profiler session initializing.
> > 2021-05-27 15:13:22.496957: I tensorflow/core/profiler/lib/profiler_session.cc:141] Profiler session started.
> > 2/221 [..............................] - ETA: 24:31 - loss: 1.7655 - accuracy: 0.2812  2021-05-27 15:13:23.337160: I tensorflow/core/profiler/lib/profiler_session.cc:66] Profiler session collecting data.
> > 2021-05-27 15:13:23.338859: I tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1743] CUPTI activity buffer flushed

@er778899789 I have the same problem as you，how did you solve it please?

**Comment by er778899789** [comment]
@Minxiangliu @yangzhenze1998 
Do you get any error in log?
ex. Some dll file not found……
Or can you guys tell me which tensorflow version you used?

By the way, I remember I add this command in my training code like below. You guys can try it.

> tf.disable_eager_execution()
